`follow_root_links()` in `WalkDir`.

Byron opened the pull request:
With it it's possible to control whether symlinks in the traversal root are followed, while defaulting to 'true' like before, or if they are handled like ordinary links.

Created in response to [this `cargo` issue](https://github.com/rust-lang/cargo/pull/11634).

Byron commented:
@BurntSushi I'd appreciate approval for the workflow to see if CI works. I did run it on my own fork but it seems to sometimes fail with rustup.exe now on Windows. However, I hope I will be luckier here. Thank you.

BurntSushi commented:
Ug right. I really really hate the "Approve and Run" thing.

BurntSushi commented:
The MSRV for walkdir is _really_ old and I would have no problems bumping it to a newer version. But if it's just for simple/small things like `[true, false]` not implementing `IntoIterator`, then I'd prefer just leaving the MSRV alone for now.

Also, I see some lines that are over 80 columns because of long strings, and rustfmt refuses to wrap those automatically. So you might need to do it manually. :)

Byron commented:
> Ug right. I really really hate the "Approve and Run" thing.

I think I addressed the issues, but… it's needing approval *again* 🤦. I do know that you can change this behaviour in the settings of the project though.

<img width="831" alt="Screenshot 2023-02-06 at 14 36 21" src="https://user-images.githubusercontent.com/63622/216985350-ee9076d8-f0d0-42dd-9aad-5368d250c499.png">

Maybe this can help to make it a little less cumbersome for all of us. It feels wrong to relegate you to clicking a button, really.

BurntSushi commented:
Ooooo!!!! I didn't know about that. Sucks I have to set it for every repo, but I've just made it as loose as possible.

Byron commented:
Great to hear I could help :)!

All tests are green now, in case you have any notes for me.

By the way, please feel free to make changes as you see fit yourself and push directly into this branch (`gh pr checkout 170` sets you up for that). That way reviews tend to be way less taxing and require less (and often no) turnarounds - at least that's how I do them in my own repos.

BurntSushi commented:
Thanks for the ping. This PR is on crates.io in `walkdir 2.4.0`.

flokli commented:
Thanks!
